src/Data/List/NonEmpty/Extra.hs

```diff
@@ -62,6 +64,29 @@ sortOn f = fromList . List.sortOn f . toList
 union :: Eq a => NonEmpty a -> NonEmpty a -> NonEmpty a
 union = unionBy (==)
 
+-- | @nubOrd@ for 'NonEmpty'. Behaves the same as 'Data.List.Extra.nubOrd'.
+--
+-- > Data.List.NonEmpty.Extra.nubOrd (1 :| [2, 3, 3, 4, 1, 2]) == 1 :| [2, 3, 4]
+-- > \xs -> Data.List.NonEmpty.Extra.nubOrd xs == Data.List.NonEmpty.Extra.nub xs
+nubOrd :: Ord a => NonEmpty a -> NonEmpty a
+nubOrd = nubOrdBy compare
+
+-- | @nubOrdBy@ for 'NonEmpty'. Behaves the same as 'Data.List.Extra.nubOrdBy'.
+--
+-- > Data.List.NonEmpty.Extra.nubOrdBy (compare `on` Data.List.length) ("a" :| ["test","of","this"]) == "a" :| ["test","of"]
+nubOrdBy :: (a -> a -> Ordering) -> NonEmpty a -> NonEmpty a
+nubOrdBy cmp (x :| xs) = x :| List.nubOrdBy cmp (List.filter (\y -> cmp x y /= EQ) xs)
```

Comment:
Is this potentially slower than nubOrdBy since it has to do O(n) comparisons for the first one, but the other one just builds a binary tree? Not slower in complexity, but a lot more comparisons. Why not just List.nubOrdBy and unsafely demand it's non'emtpy (as it always will be!) after the fact?

Reply:
Good point. By just building a binary tree every other element does not need to be compared against the first.

Reply:
I basically copied the implementation from `Data.List.NonEmpty.nubBy`, but `nubBy` is O(n^2) and every other element does need to be compared against the first anyway.